Rakefile

```diff
@@ -12,14 +14,28 @@ def collect_task_errors(tasks)
   errors.count.positive? && raise("The following tasks failed #{errors}")
 end
 
+def base_cucumber_path
+  "artifacts/#{ENV.fetch('BROWSER', 'unknown')}/#{ENV.fetch('BROWSERSTACK_CONFIGURATION_OPTIONS', 'grid')}"
```

Comment:
Sorry I meant on the second one. Not the first one. The browser is always set. Browserstack config options are not always set.

I think this needs altering a bit as well in a sub-method. Because even if the options are set. They are ignored if the browserstack var isn't set. Maybe something like

```
if browserstack?
  "artifacts/#{ENV.fetch('BROWSER')}/#{ENV.fetch('BROWSERSTACK_CONFIGURATION_OPTIONS')}"
elsif grid?
  "artifacts/#{ENV.fetch('BROWSER')}/grid"
else
    "artifacts/#{ENV.fetch('BROWSER')}/local"
end
```

Reply:
Not sure about the grid? and browserstack? functions being available in cucumber.yml and the Rakefile, but I can use the backing environment variables instead.

Are the BROWSERSTACK_CONFIGURATION_OPTIONS still optional, even if browserstack is enabled? Seems like something that we would always need to specify

Reply:
the config options are mandatory if on browserstack. but not otherwise.

Reply:
ok, so if we're $BROWSERSTACK==true, BROWSERSTACK_CONFIGURATION_OPTIONS will be set, so that's a decent enough proxy for browserstack? for use in the folder path.

Is there any reason why we can't just drop reports run by the dev on their desktop into a folder named grid inside the relevant browser folder? They are basically running the grid tests at that point, right?

Reply:
not quite. I often run locally. Which just tunnels directly to the driver on your machine, and runs your own instance of chrome/firefox

However, bunching up grid tests and local tests doesn't feel too much of a problem. So we could just call them all "non-browserstack" maybe?

Reply:
Ah, and they are the same tests running against a local driver, in the grid and in browserstack, aren't they?

Fair enough, I guess local might be the better name then.

Reply:
Kind of. Essentially selenium runs in 2 ways. Local / Remote.

But for us we have 2 "kinds" of remote. We've got a remote we control (our grid, through docker), and a remote we don't control (Browserstack).

Reply:
We've gone with "other" as the final choice.